core/src/main/scala/kafka/server/SharedServer.scala

```diff
@@ -60,7 +60,10 @@ class StandardFaultHandlerFactory extends FaultHandlerFactory {
     action: Runnable
   ): FaultHandler = {
     if (fatal) {
-      new ProcessExitingFaultHandler(action)
+      new ProcessTerminatingFaultHandler.Builder()
+        .setHalt(true)
```

Comment:
nit: `setHalt` reads a bit awkward to me. How about `shouldHalt` or `setHaltOnTerminate`?

Reply:
I agree. I went with `setShouldHalt`.